syntax = "proto3";

package auth;

service AuthService {
  rpc Register(RegisterRequest) returns (UserResponse);
  rpc Login(LoginRequest) returns (TokenResponse);
  rpc GetCurrentUser(TokenRequest) returns (UserResponse);
  rpc HealthCheck(HealthCheckRequest) returns (HealthCheckResponse);
}

message RegisterRequest {
  string email = 1;
  string password = 2;
  string full_name = 3;
}

message LoginRequest {
  string username = 1;
  string password = 2;
}

message TokenRequest {
  string token = 1;
}

message TokenResponse {
  string access_token = 1;
  string token_type = 2;
}

message UserResponse {
  int32 id = 1;
  string email = 2;
  string full_name = 3;
  bool is_active = 4;
  bool is_admin = 5;
  string created_at = 6;
  string updated_at = 7;
}

message HealthCheckRequest {
}

message HealthCheckResponse {
  string status = 1;
  string service = 2;
} 